Different csv headers by bench mode on infra error

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #103135
* __->__ #103134

As title. The headers are different for distinct bench mode. This PR is a supplement
to https://github.com/pytorch/pytorch/pull/100372 to respect `performance` mode where numerical speedup is expected
instead of status text.

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @aakhundov

BowenBao committed:
Different csv headers by bench mode on infra error

[ghstack-poisoned]

thiagocrepaldi reviewed (commented):
Please consider adding description to the PR. The ghstack does not tell why the PR is needed and how the problem is solved

BowenBao commented on benchmarks/dynamo/common.py:
```diff
@@ -2743,12 +2743,27 @@ def detect_and_mark_batch(t):
+                        [device, name, placeholder_batch_size, 0.0, 0.0]
+                        for device in args.devices
+                    ]
+                else:
+                    headers = []
+                    rows = [
+                        [device, name, placeholder_batch_size, 0.0]
+                        for device in args.devices
+                    ]
```
Headers were added by previous commit. Before that it doesn't. I'm happy to drop this branch if things are already adapted to the new format. However the branch on "performance" is important as without it we get wrong header when first model for perf test fails.

thiagocrepaldi replied:
no preference for me. I was just checking whether an unexpected behavior was being introduced

thiagocrepaldi reviewed (approved):
LGTM

BowenBao commented:
@ezyang please kindly take a look.

BowenBao committed:
Update on "Different csv headers by bench mode on infra error"


As title. The headers are different for distinct bench mode. This PR is a supplement
to https://github.com/pytorch/pytorch/pull/100372 to respect `performance` mode where numerical speedup is expected
instead of status text.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy

[ghstack-poisoned]

ezyang reviewed (approved):
oh, nice catch, thanks. I guess we should probably refactor this code eventually..

ezyang commented:
@pytorchbot merge

BowenBao commented:
@pytorchbot merge

BowenBao committed:
Update on "Different csv headers by bench mode on infra error"


As title. The headers are different for distinct bench mode. This PR is a supplement
to https://github.com/pytorch/pytorch/pull/100372 to respect `performance` mode where numerical speedup is expected
instead of status text.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy aakhundov

[ghstack-poisoned]

BowenBao committed:
Update on "Different csv headers by bench mode on infra error"


As title. The headers are different for distinct bench mode. This PR is a supplement
to https://github.com/pytorch/pytorch/pull/100372 to respect `performance` mode where numerical speedup is expected
instead of status text.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy aakhundov

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge
